From a05474967b6bdd17f4c44cea8b618e54a3ac2b3e Mon Sep 17 00:00:00 2001
From: Enrico Berti <hello@enricoberti.com>
Date: Tue, 9 Dec 2014 20:41:55 +0100
Subject: [PATCH 0133/1173] HUE-2479 [fb] Cancel moving a file with drag and
 drop

Implemented like Dropbox: if you try to copy it over itself it displays a warning
(cherry picked from commit cabed17288b8e75dd89a869f7d87541bfb7e554e)
---
 .../filebrowser/static/css/listdir_components.css  |    4 --
 .../filebrowser/templates/listdir_components.mako  |   54 ++++++++++++--------
 2 files changed, 32 insertions(+), 26 deletions(-)

diff --git a/apps/filebrowser/src/filebrowser/static/css/listdir_components.css b/apps/filebrowser/src/filebrowser/static/css/listdir_components.css
index 9557dfb..1332032 100644
--- a/apps/filebrowser/src/filebrowser/static/css/listdir_components.css
+++ b/apps/filebrowser/src/filebrowser/static/css/listdir_components.css
@@ -233,10 +233,6 @@
   border-right: 3px solid #338bb8;
 }
 
-[draggable]:hover {
-  cursor: move;
-}
-
 .progress .bar-upload {
   background-color: #338BB8;
 }
\ No newline at end of file
diff --git a/apps/filebrowser/src/filebrowser/templates/listdir_components.mako b/apps/filebrowser/src/filebrowser/templates/listdir_components.mako
index db8898c..8b76b7e 100644
--- a/apps/filebrowser/src/filebrowser/templates/listdir_components.mako
+++ b/apps/filebrowser/src/filebrowser/templates/listdir_components.mako
@@ -522,7 +522,8 @@ from django.utils.translation import ugettext as _
             }
             _dragged = ko.utils.unwrapObservable(valueAccessor().value);
           },
-          cursor: 'default'
+          cursor: "move",
+          delay: 300
         };
         dragElement.draggable(dragOptions).disableSelection();
       }
@@ -1012,35 +1013,44 @@ from django.utils.translation import ugettext as _
       self.move = function (mode) {
         var paths = [];
 
+        var isMoveOnSelf = false;
         $(self.selectedFiles()).each(function (index, file) {
+          if (file.path == $('#moveDestination').val()){
+            isMoveOnSelf = true;
+          }
           paths.push(file.path);
         });
 
-        hiddenFields($("#moveForm"), "src_path", paths);
+        if (!isMoveOnSelf){
+          hiddenFields($("#moveForm"), "src_path", paths);
 
-        $("#moveForm").attr("action", "/filebrowser/move?next=${url('filebrowser.views.view', path=urlencode('/'))}" + "." + self.currentPath());
+          $("#moveForm").attr("action", "/filebrowser/move?next=${url('filebrowser.views.view', path=urlencode('/'))}" + "." + self.currentPath());
 
-        if (mode === 'nomodal') {
-          $.jHueNotify.info('Items moving to "' + $('#moveDestination').val() + '"');
-          $("#moveForm").submit();
-        } else {
-          $("#moveModal").modal({
-            keyboard: true,
-            show: true
-          });
+          if (mode === 'nomodal') {
+            $.jHueNotify.info('${ _('Items moving to') } "' + $('#moveDestination').val() + '"');
+            $("#moveForm").submit();
+          } else {
+            $("#moveModal").modal({
+              keyboard: true,
+              show: true
+            });
 
-          $("#moveModal").on("shown", function () {
-            $("#moveModal .modal-footer div").show();
-            $("#moveHdfsTree").remove();
-            $("<div>").attr("id", "moveHdfsTree").appendTo($("#moveModal .modal-body"));
-            $("#moveHdfsTree").jHueHdfsTree({
-              home: viewModel.currentPath(),
-              onPathChange: function (path) {
-                $("#moveDestination").val(path);
-                $("#moveNameRequiredAlert").hide();
-              }
+            $("#moveModal").on("shown", function () {
+              $("#moveModal .modal-footer div").show();
+              $("#moveHdfsTree").remove();
+              $("<div>").attr("id", "moveHdfsTree").appendTo($("#moveModal .modal-body"));
+              $("#moveHdfsTree").jHueHdfsTree({
+                home: viewModel.currentPath(),
+                onPathChange: function (path) {
+                  $("#moveDestination").val(path);
+                  $("#moveNameRequiredAlert").hide();
+                }
+              });
             });
-          });
+          }
+        }
+        else {
+          $.jHueNotify.warn("${ _('You cannot copy a folder into itself.') }");
         }
       };
 
-- 
1.7.9.5

